Custom Jose4J validators don't propagate step-up authentication challenge details to WWW-Authenticate header

### Describe the bug

When using custom Jose4J validators with OIDC bearer token authentication, the `WWW-Authenticate `response header doesn't include the step-up authentication challenge details (error, error_description, acr_values) as it does when using `@AuthenticationContext` annotation.

### Expected behavior

The response should contain the full step-up authentication challenge:

`WWW-Authenticate: Bearer error="insufficient_user_authentication", error_description="Token missing ACR claim", acr_values="gold,platinum"`

### Actual behavior

When a custom Jose4J validator throws an AuthenticationFailedException with ACR values:
`throw new AuthenticationFailedException(
    "Token missing ACR claim", 
    Map.of(OidcConstants.ACR_VALUES, "gold,platinum")
);`

The response only contains:

`WWW-Authenticate: Bearer`

### How to Reproduce?

You can go through our quarkus-test-suite repo in my branch here: https://github.com/quarkus-qe/quarkus-test-suite/compare/main...jcarranzan:quarkus-test-suite:feat/oidc-step-up-auth

And enable the test disabled: `testCustomJose4jValidatorRequiresMultipleAcr`

https://github.com/jcarranzan/quarkus-test-suite/blob/4aeb91dc000d95f0adf5270059787656d9fa3739/security/keycloak-oidc-client-extended/src/test/java/io/quarkus/ts/security/keycloak/oidcclient/extended/restclient/AbstractOidcStepUpAuthenticationIT.java#L215

Then you will see the error:
```
[ERROR] io.quarkus.ts.security.keycloak.oidcclient.extended.restclient.OidcStepUpAuthenticationIT.testCustomJose4jValidatorRequiresMultipleAcr -- Time elapsed: 0.187 s <<< FAILURE!
java.lang.AssertionError: 
1 expectation failed.
Expected header "WWW-Authenticate" was not (a string containing "insufficient_user_authentication" and a string containing "acr_values" and a string containing "gold" and a string containing "platinum"), was "Bearer". Headers are:
www-authenticate=Bearer
content-length=0
```

 

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### Quarkus version or git rev

_No response_

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

--- a/docs/src/main/asciidoc/security-oidc-bearer-token-authentication.adoc
+++ b/docs/src/main/asciidoc/security-oidc-bearer-token-authentication.adoc
@@ -1706,10 +1706,13 @@ Or, if you need more flexibility, write a <<jose4j-validator-bearer>>:
 ----
 package io.quarkus.it.oidc;
 
+import java.util.Collections;
+import java.util.List;
 import java.util.Map;
 
 import jakarta.enterprise.context.ApplicationScoped;
 
+import org.eclipse.microprofile.jwt.Claims;
 import org.jose4j.jwt.MalformedClaimException;
 import org.jose4j.jwt.consumer.JwtContext;
 import org.jose4j.jwt.consumer.Validator;
@@ -1727,9 +1730,19 @@ public class AcrValueValidator implements Validator {
     @Override
     public String validate(JwtContext jwtContext) throws MalformedClaimException {
         var jwtClaims = jwtContext.getJwtClaims();
-        if (jwtClaims.hasClaim("acr")) {
-            var acrClaim = jwtClaims.getStringListClaimValue("acr");
-            if (acrClaim.contains("myACR") && acrClaim.contains("yourACR")) {
+        var acrClaimName = Claims.acr.name();
+
+        if (jwtClaims.hasClaim(acrClaimName)) {
+            // The claim 'acr' could be a String or a list
+            List<String> acrClaimValues;
+            if (jwtClaims.isClaimValueStringList(acrClaimName)) {
+               acrClaimValues = jwtClaims.getStringListClaimValue(acrClaimName);
+            } else if (jwtClaims.isClaimValueString(acrClaimName)) {
+               acrClaimValues = List.of(jwtClaims.getStringClaimValue(acrClaimName));
+            } else {
+                throw new MalformedClaimException("Claim '" + acrClaimName + "' is not a String or List of Strings.");
+            }
+            if (acrClaimValues.contains("myACR") && acrClaimValues.contains("yourACR")) {
                 return null;
             }
         }

